cpp/src/parquet/arrow/reader.h

```diff
@@ -191,11 +191,19 @@ class PARQUET_EXPORT FileReader {
   ///
   /// \returns error Status if either row_group_indices or column_indices
   ///     contains an invalid index
+  /// \deprecated Deprecated in 20.0.0. Use arrow::Result version instead.
```

Comment:
It should be 21.0.0 (if we can merge this before the code freeze; otherwise 22.0.0)